src/selectors/entities/channels.js

```diff
@@ -653,6 +653,31 @@ export const getSortedDirectChannelWithUnreadsIds = createIdsSelector(
     }
 );
 
+// getDirectAndGroupChannels returns all direct and group channels, even if they have been manually
+// or automatically closed.
+//
+// This is similar to the getDirectChannels above (which actually also returns group channels,
+// but suppresses manually closed group channels but not manually closed direct channels.) This
+// method does away with all the suppression, since the webapp client downstream uses this for
+// the channel switcher and puts such suppressed channels in a separate category.
+export const getDirectAndGroupChannels = createSelector(
+    getCurrentUser,
+    getUsers,
+    getAllChannels,
+    getTeammateNameDisplaySetting,
+    (currentUser, profiles, channels, settings) => {
+        if (!currentUser) {
+            return [];
+        }
+
+        return Object.values(channels).filter((channel) =>
```

Comment:
@lieut-data Can you add a minor check for channelObj similar to https://github.com/mattermost/mattermost-redux/pull/588. we still are not sure why undefined channel objects get into the state this is just as a precaution